sdk/metric/controller/basic/controller.go

```diff
@@ -216,23 +249,29 @@ func (c *Controller) collect(ctx context.Context) error {
 // compute the CheckpointSet.  This applies the configured collection
 // timeout.  Note that this does not try to cancel a Collect or Export
 // when Stop() is called.
-func (c *Controller) checkpoint(ctx context.Context, cond func() bool) error {
-	ckpt := c.checkpointer.CheckpointSet()
+func (c *Controller) checkpoint(ctx context.Context) error {
+	for _, impl := range c.provider.List() {
+		if err := c.checkpoint1(ctx, impl.(*accumulatorCheckpointer)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Controller) checkpoint1(ctx context.Context, ac *accumulatorCheckpointer) error {
```

Comment:
Can we rename this to better describe the function it performs.

Reply:
Changed to 
```
// checkpointSingleAccmulator checkpoints a single instrumentation
// library's accumulator, which involves calling
// checkpointer.StartCollection, accumulator.Collect, and
// checkpointer.FinishCollection in sequence.
func (c *Controller) checkpointSingleAccmulator(ctx context.Context, ac *accumulatorCheckpointer) error {
```